### ISSUE TITLE
Issue cross-referencing section headers

### ISSUE THREAD

**Issue by ssp3nc3r** [rstudio/distill#287]
The following `\@ref()` displays `??` but does correctly link to the heading when clicked:

```
# heading 1 {#myheading1}

See section \@ref(myheading1)
```

Distill version: 1.1.2, GithubSHA1: e53e8ff1e5608231eace6264df4a33a1b81f2876

**Comment by ssp3nc3r** [comment]
It seems the issue is related to the ability to number headings in distill documents, because the reference `\@ref()` would normally output the number of the section.

In a related issue, I have been able to get numbered sections to work in a distill document specified as a base format for `bookdown::html_document2` and including `pandoc_args: ["--number-sections"]`, see https://github.com/rstudio/distill/issues/56#issuecomment-766136237, but that wasn't enough to get `??` replaced with the numbered section.

**Comment by ssp3nc3r** [comment]
Anyone know of a workaround or hack for this until there's a fix that's easier than opening the html and manually replacing the ?? with the numbers?

**Comment by cderv** [comment]
This does not currently work with **bookdown** feature because the HTML from **distill** as some special attributes on `h*` tags that **bookdown** does not expect. 

This should be a small fix in **bookdown** `parse_section_labels()` internal function to support the trick of 
````yaml
output:
  bookdown::html_document2:
    base_format: distill::distill_article
    pandoc_args: ["--number-sections"]
```

**Comment by cderv** [comment]
I look a bit more and in fact this is not supported in **bookdown** because the feature currently expects sections and `html4` and **distill** produces `html5` and don't have sections (which could be added with `--section-divs` in **distill**) - this would require some adjustment. 

Having the referencing feature outside of **bookdown** would help for this.

**Comment by ssp3nc3r** [comment]
> I look a bit more and in fact this is not supported in **bookdown** because the feature currently expects sections and `html4` and **distill** produces `html5` and don't have sections (which could be added with `--section-divs` in **distill**) - this would require some adjustment.
> 
> Having the referencing feature outside of **bookdown** would help for this.

Thanks for looking into this. By

> could be added with `--section-divs` in **distill**

would that be a pandoc argument or something? Or is it more customized than that?

**Comment by cderv** [comment]
> would that be a pandoc argument or something? Or is it more customized than that?

Yes it is pandoc argugment, but I believe distill in HTML5 and **bookdown** does not really handles that yet. I did not try to play with it yet.

**Comment by ssp3nc3r** [comment]
Ok, so doing this,


````
---
output:
  bookdown::html_document2:
    base_format: distill::distill_article
    pandoc_args: ["--number-sections", "--section-divs"]
---

# Section

See section \@ref(section)
````

Does create sections for headers in the html:

```
<d-article>
<section id="section" class="level1" data-number="1">
<h1 data-number="1"><span class="header-section-number">1</span> Section</h1>
<p>See section <a href="#section"><strong>??</strong></a></p>

</section>
<!--radix_placeholder_article_footer-->
<!--/radix_placeholder_article_footer-->
</d-article>
```

But as seen in the html above, the reference works but shows ??.

1 Section
See section ??

**Comment by cderv** [comment]
Unfortunately, it is not as simple. As I said other adjustments are required for this to work.
This is because `--section-divs` will not output the same in HTML4 and HTML5 - **bookdown** expect HTML4 and **distill** is HTML5. Some work need to be done in **bookdown** to support it. 

We'll look into that in the future. Thanks for your patience.
